src/flp/types/fixedpoint_l2.rs

```diff
@@ -0,0 +1,816 @@
+        // the polynomial is:
+        //   p(y) = 2^(2n-2) + -(2^n) * y + 1 * y^2
+        let linear_part = fi_one << (fi_bits_per_entry);
+        let constant_part = fi_one << (fi_bits_per_entry + fi_bits_per_entry - fi_two);
+        let norm_summand_poly = vec![F::from(constant_part), -F::from(linear_part), F::one()];
+
+        // Compute chunk length and number of calls for parallel sum gadgets.
+        let len0 = bits_per_entry * entries + bits_for_norm;
+        let gadget0_chunk_len = std::cmp::max(1, (len0 as f64).sqrt() as usize);
+        let mut gadget0_calls = len0 / gadget0_chunk_len;
+        if len0 % gadget0_chunk_len != 0 {
+            gadget0_calls += 1;
+        }
+
+        let len1 = entries;
+        let gadget1_chunk_len = std::cmp::max(1, (len1 as f64).sqrt() as usize);
+        let mut gadget1_calls = len1 / gadget1_chunk_len;
+        if len1 % gadget1_chunk_len != 1 {
+            gadget1_calls += 1;
+        }
+
+        Ok(Self {
+            bits_per_entry,
+            fi_bits_per_entry,
+            entries,
+            bits_for_norm,
+            range_01_checker: poly_range_check(0, 2),
+            norm_summand_poly,
+            phantom: PhantomData,
+
+            // range constants
+            range_norm_begin: entries * bits_per_entry,
+            range_norm_end: entries * bits_per_entry + bits_for_norm,
+
+            // configuration of parallel sum gadgets
+            gadget0_calls,
+            gadget0_chunk_len,
+            gadget1_calls,
+            gadget1_chunk_len,
+        })
+    }
+}
+
+impl<
+        T: Fixed,
+        F: FieldElement,
+        SPoly: ParallelSumGadget<F, PolyEval<F>>,
+        SBlindPoly: ParallelSumGadget<F, BlindPolyEval<F>>,
+    > Clone for FixedPointBoundedL2VecSum<T, F, SPoly, SBlindPoly>
+{
+    fn clone(&self) -> Self {
+        Self {
+            bits_per_entry: self.bits_per_entry,
+            fi_bits_per_entry: self.fi_bits_per_entry,
+            entries: self.entries,
+            bits_for_norm: self.bits_for_norm,
+            range_01_checker: self.range_01_checker.clone(),
+            norm_summand_poly: self.norm_summand_poly.clone(),
+            phantom: PhantomData,
+
+            // range constants
+            range_norm_begin: self.range_norm_begin,
+            range_norm_end: self.range_norm_end,
+
+            // configuration of parallel sum gadgets
+            gadget0_calls: self.gadget0_calls,
+            gadget0_chunk_len: self.gadget0_chunk_len,
+            gadget1_calls: self.gadget1_calls,
+            gadget1_chunk_len: self.gadget1_chunk_len,
+        }
+    }
+}
+
+impl<T, F, SPoly, SBlindPoly> Type for FixedPointBoundedL2VecSum<T, F, SPoly, SBlindPoly>
+where
+    T: Fixed + CompatibleFloat<F>,
+    F: FieldElement,
+    SPoly: ParallelSumGadget<F, PolyEval<F>> + Eq + 'static,
+    SBlindPoly: ParallelSumGadget<F, BlindPolyEval<F>> + Eq + 'static,
+{
+    const ID: u32 = 0xFFFF0000;
```

Comment:
This codepoint is used for another type, so we'll need a different one. (cc/ @divergentdave, who might have an opinion about what point to allocate).

Relatedly, we should write unit tests to ensure that the codepoints for all of our "private" VDAFs are distinct.  This is separate from this PR, however.

Reply:
We currently use `0xFFFF0000` in six places, including Prio3Aes128Average, Prio3Aes128CountVec, Poplar with ToyIDPF, and Prio v2. I may have misunderstood you in https://github.com/divviup/libprio-rs/pull/289#discussion_r956228387, should we be using distinct codepoints for other non-standardized VDAFs as well? (If so, that'll call for another issue/PR)

Reply:
You did not misunderstand me, I'm contradicting my past self! (Sorry about that.) Let's leave this here for now and we'll follow up in the issue you filed.